hack/lib/util.sh

```diff
@@ -437,21 +437,35 @@ kube::util::ensure_clean_working_dir() {
 
 # Ensure that the given godep version is installed and in the path.  Almost
 # nobody should use any version but the default.
+#
+# Sets:
+#  KUBE_GODEP: The path to the godep binary
+#
 kube::util::ensure_godep_version() {
-  GODEP_VERSION=${1:-"v79"} # this version is known to work
+  local godep_target_version=${1:-"v79"} # this version is known to work
 
-  if [[ "$(godep version 2>/dev/null)" == *"godep ${GODEP_VERSION}"* ]]; then
+  # If KUBE_GODEP is already set, and it's the right version, then use it.
+  if [[ -n "${KUBE_GODEP:-}" && "$(${KUBE_GODEP:?} version 2>/dev/null)" == *"godep ${godep_target_version}"* ]]; then
```

Comment:
why do we need to allow this?  I'd rather just keep this clean if we can.  Do we have a case where a user would really want to override this?

Reply:
This isn't for users necessarily to override, but it's so that if we've already done the go install from vendor once, *and* it's the right version, that we use it. The primary need for this is in the verify-godeps script where we modify GOPATH to a temp folder. The save step errors if we try to reinstall godep inside the temp GOPATH. I didn't dig into it enough to figure out *exactly* what goes on, but I figured it was best to not pollute the gopath that we're using to verify deps anyways. And we save whatever CPU cycles it would take to compile/install from vendor the two subsequent times.

Reply:
Can we get https://github.com/kubernetes/kubernetes/pull/57600/commits/55cca04439b4a476015ef8697fff8b79df8d49f5#diff-fa58360a7125fddab9491c80bb3f1a87 in first?  I am not sure I see where the problem is...